Kernel/Bus/USB/UHCI/UHCIController.cpp

```diff
@@ -510,6 +519,41 @@ ErrorOr<size_t> UHCIController::submit_bulk_transfer(Transfer& transfer)
     return transfer_size;
 }
 
+ErrorOr<void> UHCIController::submit_async_interrupt_transfer(NonnullLockRefPtr<Transfer> transfer, u16 ms_interval)
+{
+    dbgln_if(UHCI_DEBUG, "UHCI: Received interrupt transfer for address {}. Root Hub is at address {}.", transfer->pipe().device_address(), m_root_hub->device_address());
+
+    if (ms_interval == 0) {
+        return EINVAL;
+    }
+
+    SpinlockLocker locker(m_async_lock);
+
+    auto transfer_queue = TRY(create_transfer_queue(*transfer));
+
+    u8 idx = 0;
+    while ((1 << idx) < ms_interval && idx < NUMBER_OF_INTERRUPT_QHS) {
+        idx++;
+    }
+
+    m_active_async_transfers.append({ transfer, transfer_queue, ms_interval });
```

Comment:
move() the transfer here perhaps? This should be `TRY(try_append()` at the very least, which then makes it obvious that it's another possible allocation under a spinlock (due to vector resize) :/

Perhaps this should be a FixedArray with a size set at init? At least we should apply a bound to this vector/queue somehow before appending

Reply:
Yeah the Vector could be replaced with FixedArray<AsyncTransferHandle>(MAXIMUM_NUMBER_OF_QHS) then we can avoid dynamic memory altogether.